packages/vscode-zipfs/rollup.config.js

```diff
@@ -1,27 +1,43 @@
-import cjs     from '@rollup/plugin-commonjs';
-import resolve from '@rollup/plugin-node-resolve';
-import path    from 'path';
-import esbuild from 'rollup-plugin-esbuild';
+import cjs            from '@rollup/plugin-commonjs';
+import resolve        from '@rollup/plugin-node-resolve';
+import path           from 'path';
+import esbuild        from 'rollup-plugin-esbuild';
+import {defineConfig} from 'rollup';
 
 const mode = process.env.NODE_ENV || `production`;
 
 // eslint-disable-next-line arca/no-default-export
-export default async () => ({
-  input: `./sources/index.ts`,
-  output: {
-    file: `./build/index.js`,
-    format: `cjs`,
-  },
-  plugins: [
-    resolve({
-      extensions: [`.mjs`, `.js`, `.ts`, `.tsx`, `.json`],
-      rootDir: path.join(__dirname, `../../`),
-      jail: path.join(__dirname, `../../`),
-      preferBuiltins: true,
-    }),
-    esbuild({tsconfig: false, target: `node12`}),
-    cjs({transformMixedEsModules: true, extensions: [`.js`, `.ts`]}),
-    mode === `production` && (await import(`rollup-plugin-terser`)).terser({ecma: 2019}),
-  ],
-  external: [`vscode`],
-});
+export default async () =>
+  defineConfig({
+    treeshake: {
+      propertyReadSideEffects: false,
+      unknownGlobalSideEffects: false,
+      tryCatchDeoptimization: false,
+    },
```

Comment:
I'm worried about those, as it would be a massive pain to debug any issue they could accidentally cause. The gain seems fairly low, so I'd tend to favour long-term maintainability.

Reply:
That's fair, reverted 👍